lib/Dialect/SV/Transforms/SVExtractTestCode.cpp

```diff
@@ -609,6 +675,21 @@ void SVExtractTestCodeImplPass::runOnOperation() {
   this->instanceGraph = &getAnalysis<circt::hw::InstanceGraph>();
 
   auto top = getOperation();
+
+  // It takes extra effort to inline modules which contains inner symbols
+  // referred through hierpaths or unknown operations since we have to update
+  // inner refs users globally. However we do want to inline modules which
+  // contain bound instances so create a set of inner refs used by non bind op
+  // in order to allow bind ops.
+  DenseSet<hw::InnerRefAttr> innerRefUsedByNonBindOp;
+  top.walk([&](hw::InnerRefUserOpInterface op) {
+    if (!isa<sv::BindOp>(op))
+      for (auto attr : op->getAttrs())
+        attr.getValue().walk([&](hw::InnerRefAttr attr) {
+          innerRefUsedByNonBindOp.insert(attr);
+        });
+  });
```

Comment:
Can this be called multiple times on the same module (causing performance issues)?

Reply:
No, it called just once for each module. It's still a bit concerning  to walk the entire circuit (and its attributes) but I think there is no other way around.